### ISSUE TITLE
[Regression] Static content deploy is not possible when one of themes have no static assets

### ISSUE THREAD

**Issue by Igloczek** [magento/magento2#12217]
It's exactly same issue as #7411, confirmed by few others in comments.

It makes deployment not possible, even for built-in languages like `en_US`, while we have a theme without something in `web` directory. 

It's also kinda weird, that when we decide to compile theme from the core, like "Blank", it will also fail if one of the other themes doesn't have all "required" files.

Workaround - add some static asset i.e. `web/placeholder.js`

### Preconditions
Magento 2.1.10

### Steps to reproduce
1. Create a theme without anything in `web` folder
2. Run `bin/magento setup:static-content:deploy`

### Expected result
Successful assets compilation

### Actual result
```
  [Exception]
  Notice: Undefined offset: 5 in [...]/vendor/magento/module-deploy/Model/Deploy/LocaleDeploy.php on line 287
```

**Comment by magento-engcom-team** [comment]
@Igloczek, thank you for your report.
We've created internal ticket(s) MAGETWO-83997 to track progress on the issue.

**Comment by digvijay2017** [comment]
I have noticed this issue in ver 2.1.12. But, This issue has been fixed in ver 2.2.1.

**Comment by krzksz** [comment]
It seems that the issue was fixed in mentioned PRs, I have also confirmed that on Magento 2.3.x theme with empty `web` directory compiles properly.
